data-runtime/src/main/java/io/micronaut/data/runtime/mapper/DTOMapper.java

```diff
@@ -101,7 +112,7 @@ public Object read(@NonNull S object, @NonNull Argument<?> argument) {
      * @param property THe property
      * @return The result
      */
-    public @Nullable Object read(@NonNull S resultSet, @NonNull RuntimePersistentProperty<T> property) {
+    public @Nullable Object read(@NonNull S resultSet, @NonNull RuntimePersistentProperty<?> property) {
```

Comment:
Why is this change necessary? If the association is being queried we probably should instantiate another `DTOMapper` with the appropriate entity